### PR TITLE
added support for macOS and windows via cross platform open (cxopen)

### PR DESCRIPTION
**Pull request by isubasinghe**

### CODE REVIEW

File: dot_graph.py [hunk 1]
```diff
@@ -11,7 +11,9 @@
 
 import sys
 import os
-import source
+from typing import TypeVar, Mapping
```
Comment (math2001) [inline dot_graph.py]: did you mean these typing imports?

File: dot_graph.py [hunk 2]
```diff
@@ -11,7 +11,9 @@
...
+import cxopen
+from source import Expr, ExprNum, ExprOp, ExprVar, Function, NodeBasic, NodeCall, NodeCond, NodeEmpty, Operator, Update, convert_function
```
Comment (math2001) [inline dot_graph.py]: I think a lot things use `source.` in this file, so I don't think you meant that either

**Review by math2001 — CHANGES_REQUESTED** [review]
Good idea :) Just a few issues with imports

**Comment by isubasinghe** [comment]
Ah I didn't make these changes myself, I'm assuming my editor did.  I've removed those changes now. 
Sorry, I'll check the diff next time.

File: cxopen.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# cross platform support for opening files (for example images)
+import sys
+
+
+def get_command_for_open():
```
Comment (math2001) [inline cxopen.py]: forgot to mention this before, the type checker isn't happy (`mypy --strict .`), it wants every function to have a type definition

Comment (isubasinghe) [reply cxopen.py]: Ah sorry, that is also addressed now. I am working on getting the pre-commit installed (had a bit of trouble with my nix installation of Python, I think I found a solution).

**Comment by isubasinghe** [comment]
Gonna convert this to draft until I can actually run this on my machine so I don't bother you with broken PRs

**Comment by zaklogician** [comment]
Whos' gonna be able to review this? I don't know anybody who runs Windows around here :)

**Pull request closed without merging**